lib/nfc/helpers/gallagher_util.c

```diff
@@ -0,0 +1,30 @@
+/* gallagher_util.c - Utilities for parsing Gallagher cards (New Zealand).
+ * Author: Nick Mooney (nick@mooney.nz)
+ * 
+ * Reference: https://github.com/megabug/gallagher-research
+*/
+
+#include "gallagher_util.h"
+
+/* Precondition: cardholder_data_obfuscated points to at least 8 safe-to-read bytes of memory.
```

Comment:
Is there anything more coding-style-wise I should do to ensure this?

Reply:
In this case, we're reading `cardholder_data_obfuscated`, which is just 8 raw bytes that need to be shuffled around and have the bits masked out of them to pull out a credential -- so a pointer to GallagherCredentialBlock would not be appropriate, I don't think. Let me know if you disagree.